pkg/store/gcworker/gc_worker.go

```diff
@@ -1152,22 +1152,36 @@ func (w *GCWorker) checkUsePhysicalScanLock() (bool, error) {
 }
 
 func (w *GCWorker) resolveLocks(ctx context.Context, safePoint uint64, concurrency int, usePhysical bool) (bool, error) {
+	now, err := w.getOracleTime()
+	if err != nil {
+		logutil.Logger(ctx).Error("get Oracle time failed", zap.String("category", "gc worker"),
+			zap.String("uuid", w.uuid),
+			zap.Uint64("safePoint", safePoint),
+			zap.Error(err))
+	}
+	w.checkGCInterval()
+	resolveLockPoint := oracle.ComposeTS(now.Add(-gcDefaultRunInterval).Unix()*1000, 0)
+	if safePoint > resolveLockPoint {
+		resolveLockPoint = safePoint
+	}
```

Comment:
I'm sorry, 1) you seem to be mixing the concept of GC run interval and GC lifetime; 2) you are not considering user's configuration at all, and a user don't have choice to disable this new behavior; 3) you are breakig the constraints of alive transactions and service safepoints so that long-running transactions is possible to be killed. A previously added but later reverted change to this part is to do *soft* resolving lock at now-5min (do not resolve locks no transactions whose tts is not expired yet), and do *hard*(force) resolving lock at safepoint (which is how the current GC resolving lock is being done). It looks more proper and you may consider adopt that way.
Also consider making the soft resolve lock interval configurable.

Reply:
I have to say that it is not the final version, so it doesn't have a configurable variable. I just want to know why the "soft resolve lock" reverted.

Reply:
Actually, there's another way to handle the original issue. A new SQL like `resolve locks` could be introduced for users to resolve locks on demand. However, it will increase the learning cost for users.